### ISSUE TITLE
[Bug]: Auto-distributed timetabled trains leave depot in order different from timetable requirements

### ISSUE THREAD

**Issue by CitrusWire** [OpenTTD/OpenTTD#10374]
### Version of OpenTTD

12.2

### Expected result

OpenTTD 12.2, No GRFs

Summary: Auto-distributed timetabled trains leave the depot in an order that is different from the timetable requirements, screwing everything up.

Load attached game
* Pause game
* The train in the depot already has a good timetable all set up.
* Ctrl-Clone it to get 10 trains.
* Select "train # 1", Ctrl-click "Start date" on the timetable, and set a start date of august 30th.
* Start all trains. Unpause game.
* Train 2 leaves the depot second and immediately shows as being 401 days early!

The reason is that train 10 is the second train in this particular timetabled chain of trains, but train 2 leaves the depot first and fills the only remaining platform, then train 3, etc. You can test this by connecting the other platforms and watching them properly self-distribute.

You can also mess it up by selecting train 5 (or any train that's not 1 or 2), and setting up the ctrl-click start date with them as well.

Put simply: The order the trains leave the depot is different from the order that the trains have been assigned to the timetable. This doesn't always happen, it's basically random - my first test using the above methodology all the trains went fine.

I'd suggest the fix (from a make-it-as-easy-as-possible-for-the-player) is:
* Look at the timetables of in-depot trains
* If they're shared orders and using a timetable, then sort according to start date.

Though this will get more complicated if multiple timetables are started from the same depot at the same time.

### Actual result

As Above

### Steps to reproduce

As Above

**Comment by CitrusWire** [comment]
(Couldn't seem to attach the file to the original report.)

[Testing train auto distribution, 1952-08-14.zip](https://github.com/OpenTTD/OpenTTD/files/10449957/Testing.train.auto.distribution.1952-08-14.zip)

**Comment by LC-Zorg** [comment]
> make-it-as-easy-as-possible-for-the-player

if so:
- throw all this horrible feature to the trash
- create new / adapt from JGRPP the auto separation function, which can be turned on and off with one button (simple for everyone, not just for programmers and creators of this function)

Something like this:
![Auto Separation - one button](https://user-images.githubusercontent.com/67662046/213564608-e3be4931-1904-46e0-83bd-11df23feaa1e.png)

This feature is a nightmare. It's as if someone introduced the entire procedure of launching a space rocket into an activity as simple as starting a family car. The whole procedure contains 12 steps (according to the wiki), where a mistake, omission of one step, ends as described in this report. It's complete nonsense. It should be one button (on/off) and nothing else. I don't see any advantage of this solution over auto separation. I'm wrong?

**Comment by LC-Zorg** [comment]
Out of curiosity, I wanted to check where the problem mentioned here could be and... I noticed another bug. In my game, apart from the reversed order of adding dates (the first vehicle is last, the last is first), subsequent dates went into the past instead of into the future, causing some of the vehicles to be late already when they left the depot.
![Automatic chaos](https://user-images.githubusercontent.com/67662046/213756834-fd67e870-e695-4aa6-acb2-74a36cc5ef46.png)

I've also checked it out in the past and it seems this bug has been present since at least 1.6.1.

**Comment by 2TallTyler** [comment]
For what it's worth, #11945 handles this properly even with timetabled trains (it re-orders them as they leave the depot), although only if you explicitly set the depot unbunching order. So this bug is not entirely fixed.
